src/common/types.cpp

```diff
@@ -665,17 +665,49 @@ bool LogicalType::IsMoreGenericThan(LogicalType &other) const {
 		return false;
 	case LogicalTypeId::DATE:
 		return false;
-	case LogicalTypeId::TIMESTAMP:
-	case LogicalTypeId::TIMESTAMP_NS:
-	case LogicalTypeId::TIMESTAMP_MS:
-	case LogicalTypeId::TIMESTAMP_SEC:
+	case LogicalTypeId::TIMESTAMP: {
+		switch (other.id()) {
+		case LogicalTypeId::TIMESTAMP_NS:
+		case LogicalTypeId::TIME:
+		case LogicalTypeId::DATE:
+			return true;
+		default:
+			return false;
+		}
+	}
+	case LogicalTypeId::TIMESTAMP_NS: {
+		switch (other.id()) {
+		case LogicalTypeId::TIME:
+		case LogicalTypeId::DATE:
+			return true;
+		default:
+			return false;
+		}
+	}
+	case LogicalTypeId::TIMESTAMP_MS: {
 		switch (other.id()) {
+		case LogicalTypeId::TIMESTAMP:
+		case LogicalTypeId::TIMESTAMP_NS:
```

Comment:
I'm not sure I understand the reasoning here? Isn't a ns representation more generic than a ms representation?

Reply:
Yes, this is wrong.

Reply:
Fix looks good!